chore: fix code quality issues

withshubh opened the pull request:
#### Description
Hi :wave: I ran the [DeepSource](https://deepsource.io) static analyzer on the forked copy of this repo and found some interesting [code quality issues](https://deepsource.io/gh/withshubh/determined/issues/?category=recommended). This PR fixes a few of them.

#### Summary of changes
- ~Remove length check in favour of truthiness of the object~
- Remove unnecessary use of comprehension
- Use literal syntax instead of function calls to create data structure
- ~added .deepsource.toml file~
- Merge `isinstance` calls

rb-determined-ai commented on .deepsource.toml:
```diff
@@ -0,0 +1,14 @@
+version = 1
```
We appreciate the config, but we are pretty happy with our current static analysis tooling.  Please omit this file from the PR.

rb-determined-ai commented on common/determined_common/api/authentication.py:
```diff
@@ -164,7 +164,7 @@ def _write_store(self) -> None:
         self._create_det_path_if_necessary()
         cache_path = self._get_token_cache_path()
         store = {}
-        if self._tokens is not None and len(self._tokens):
```
Honestly, I think because we use mypy, using on `if len(thing):` is superior to depending on the truthiness of `if thing:` because the truthiness test will sometimes hide bugs from mypy if you e.g. change `thing` to/from `Optional[ThingType]`.  So I'd prefer that these sorts of changes be omitted.

(also if we were going to depend on the truthiness test, the proposed edit is redundant anyway)

rb-determined-ai reviewed (commented):
Thank you for offering these improvements to our project!  You will have to sign a cla agreement before the PR will be accepted, which is a process you can start [here](https://determined.ai/cla).

The changes from dictionary comprehension to `dict()` look good.

Please also rename the PR to be `chore: ...` rather than `perf: ...`, as the changes proposed will not be any measurable performance impact within our project.  The computationally expensive parts of our project are all deep-learning related and would dwarf any performance impact from an extra dictionary comprehension or `len()` call.

rb-determined-ai reviewed (approved):
These changes look great, thank you!  We got your signed CLA and I'll make sure this lands tomorrow.

Thank you very much for your contribution!

withshubh commented:
@rb-determined-ai :wave: 

I have -
- Signed and e-mailed the CLA
- removed `.deepsource.toml` file
- reverted fix `Remove length check in favour of truthiness of the object`
- pushed a new commit - Merge `isinstance` calls

Please review :sparkling_heart:

rb-determined-ai commented:
@cla-bot[bot] check

rb-determined-ai commented:
@withshubh Your cla bot is good, but I think you are failing the linting step.  You can test it yourself locally by running `make -C common check`, and so you should be able to fix it and verify your fix locally if you would like.

rb-determined-ai reviewed (changes requested):
Hey, I thought you had removed the .deepsorce.toml file?  I see it here again after the latest commit.

withshubh commented:
> Hey, I thought you had removed the .deepsorce.toml file? I see it here again after the latest commit.

My bad. I pushed it by mistake from my local clone.
Deleted. :sparkling_heart:

rb-determined-ai commented:
Thank you @withshubh, I will approve and merge this change after CI passes.

withshubh commented:
@rb-determined-ai 
Also, I would really appreciate it if you could drop any suggestion on what we can do better to onboard you (or Determined-ai) on DeepSource in the future. :)

rb-determined-ai commented:
@withshubh I think you have done a fine job at dev outreach!  Honestly I was impressed you followed up after the initial PR review.  But the fact is that we are just satisfied with what we have right now, and that our team is already comfortable with.
